### ISSUE TITLE
Learn Microservices using Kubernetes and Istio Issues

### ISSUE THREAD

**Issue by ericvn** [istio/istio.io#7904]
https://preliminary.istio.io/latest/docs/examples/microservices-istio/

The main issue remaining to be fixed is none of the tutorial data is showing up in the dashboards. I only see data for the `default` namespace (namely the httpbin). Maybe something has changed in this space.

- Setup a Kubernetes Cluster - **FIXED**
Looks to expect that the add-ons are still part of the demo profile.
Minor update needed to add the add-ons to the configuration.
If you try to install them later, there are some some permissions setup in this step that may prohibit that (at least I tried to install things later and have a bunch of forbidden errors)

NOTE: I ran without actually adding on the addons here and ran into a permissions issue later. I need to start this over again adding in the addons to see if things will work (haven't done that yet). The Istio Ingres gateway page (which is the only one listed in the testing sheet) at the end of this list of pages did not work.

- Setup a Local Computer - **FIXED**
Istioctl version output:
```
istioctl version
client version: 1.7.0-beta.2
control plane version: 1.7.0-beta.2
data plane version: 1.7.0-beta.2 (4 proxies)
```

- Run a Microservice Locally - **FIXED**
npm install yields a warning:
```
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN ratings No description
npm WARN ratings No repository field.
npm WARN ratings No license field.

added 24 packages from 22 contributors and audited 24 packages in 0.935s
found 1 high severity vulnerability
  run `npm audit fix` to fix them, or `npm audit` for details
```
We should probably update the file to fix this.
```
 npm audit                                                                                                                                               ✔  14:00:58

                       === npm audit security report ===

# Run  npm install mongodb@3.6.0  to resolve 1 vulnerability
SEMVER WARNING: Recommended action is a potentially breaking change
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Denial of Service                                            │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ mongodb                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ mongodb                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ mongodb                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1203                            │
└───────────────┴──────────────────────────────────────────────────────────────┘


found 1 high severity vulnerability in 24 scanned packages
  1 vulnerability requires semver-major dependency updates.
```
Updated in package.json: `    "mongodb": "^3.6.0",`

- Run ratings in Docker - **FIXED**
We should probably remove the stopped container as well.
```
docker rm <the container ID from the output of docker ps>
```

- Run Bookinfo with Kubernetes - **FIXED**
Browser page for BookInfo isn't formatted correctly.
NOTE: Since I see the output calling v2 correctly, I suspect the formatting issue is preventing me from seeing the stars.

- Add a new version of reviews - **FIXED**
Nothing I did got me to the second version of reviews (with the stars).

- Enable Istio on productpage - **FIXED**. but maybe we don't need the command anymore)
First command fails:
```
kubectl apply -f - <<EOF
apiVersion: "authentication.istio.io/v1alpha1"
kind: Policy
metadata:
  name: default
spec:
  peers: []
EOF
kubectl apply -f https://raw.githubusercontent.com/istio/istio/release-1.7/samples/bookinfo/networking/destination-rule-all.yaml

error: unable to recognize "STDIN": no matches for kind "Policy" in version "authentication.istio.io/v1alpha1"
destinationrule.networking.istio.io/productpage unchanged
destinationrule.networking.istio.io/reviews unchanged
destinationrule.networking.istio.io/ratings unchanged
destinationrule.networking.istio.io/details unchanged
```

Couldn't use the dashboard since the addons weren't installed and one can't install them with permissions set.

- Enable Istio on all the microservices
 second command fails:  - **FIXED**
```
 curl -s https://raw.githubusercontent.com/istio/istio/release-1.7/samples/bookinfo/platform/kube/bookinfo.yaml | istioctl kube-inject -f - | kubectl apply -l app!=reviews -f -
curl -s https://raw.githubusercontent.com/istio/istio/release-1.7/samples/bookinfo/platform/kube/bookinfo.yaml | istioctl kube-inject -f - | kubectl apply -l app=reviews,version=v2 -f -

service/details unchanged
serviceaccount/bookinfo-details unchanged
deployment.apps/details-v1 configured
service/ratings unchanged
serviceaccount/bookinfo-ratings unchanged
deployment.apps/ratings-v1 configured
serviceaccount/bookinfo-reviews unchanged
service/productpage unchanged
serviceaccount/bookinfo-productpage unchanged
deployment.apps/productpage-v1 configured
The Deployment "reviews-v2" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app":"reviews", "version":"v2"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable
```

Of course, reviews-v2 is only running 1 container and there are 3.

- Configure Istio Ingress Gateway
THis step did not work.  `This istio.tutorial.bookinfo.com page can’t be found`

Again, can't do any of the dashboard items

**Comment by ericvn** [comment]
Working through this again, I noted several items:

reolved should be resolved
four services should be four pods

**Comment by ericvn** [comment]
Bookinfo image:
![image](https://user-images.githubusercontent.com/10537847/90156557-371e2280-dd52-11ea-939d-95cbb7719da4.png)

**Comment by ericvn** [comment]
Got to normal browser window changing ingress (adding pathType of prefix):
```
      - path: /static
        pathType: Prefix
        backend:
          serviceName: productpage
          servicePort: 9080
```

**Comment by ericvn** [comment]
This tutorial was fixed by several PRs: #7974 , #7941 , #7959 and https://github.com/istio/istio/pull/26593.
